### PR TITLE
bump version and update changelog

### PR DESCRIPTION
**Pull request by johrstrom**
bump version and update changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -524,7 +529,10 @@ Functionally the same as [0.17.3] but with some CI updates.
 ### Added
 - Initial release!
 
-[Unreleased]: https://github.com/OSC/ood_core/compare/v0.23.4...HEAD
+[Unreleased]: https://github.com/OSC/ood_core/compare/v0.24.2...HEAD
+[0.24.2]: https://github.com/OSC/ood_core/compare/v0.24.2...v0.24.1
+[0.24.1]: https://github.com/OSC/ood_core/compare/v0.24.1...v0.24.0
+[0.24.0]: https://github.com/OSC/ood_core/compare/v0.24.0...v0.23.5
```
Comment (Oglopf) [inline CHANGELOG.md:535]: Are those version numbers at the end right? My brain wants these the other way `v0.23.5...v0.24.0` etc. Am I reading that wrong?

Comment (johrstrom) [reply CHANGELOG.md:535]: Yep! You're right, they're backwards.

Comment (johrstrom) [reply CHANGELOG.md]: and oddly enough, it was just that one link.

Comment (johrstrom) [reply CHANGELOG.md]: Wait, now I'm all mixed up. It should be `lower...higher`. They're links, so they should actually link to something correct. I'm validating them all now.

Comment (johrstrom) [reply CHANGELOG.md]: OK - I think they're all correct now.

**Review by Oglopf — APPROVED** [review]
Looks good.

**Pull request merged**